src/main/java/net/foxes4life/foxclient/gui/BlockHud.java

```diff
@@ -0,0 +1,120 @@
+
+    public BlockHud(MinecraftClient client) {
+        this.client = client;
+    }
+
+    public void render(DrawContext context, ItemRender itemRender) {
+        Block block = getBlock();
+        boolean show = false;
+
+        if (block != null && !(block instanceof AirBlock)) {
+            lastBlock = block;
+            show = true;
+        }
+        else {
+            block = lastBlock;
+        }
+
+        // if the block is still null, we don't want to render anything
+        if (block == null) return;
+
+        boolean showAnimations = Main.config.get(FoxClientSetting.BlockHudAnimations, Boolean.class);
+
+        float delta = client.getTickDelta();
+
+        final int padding = 2;
+        final int border = 3;
+        final int height = 16;
+
+        final int minY = -height - padding * 2 - border - 2;
+        int goalY = show ? 0 : minY;
+
+        // transition y position
+        if (showAnimations) y = MathHelper.lerp(.6f * delta, y, goalY);
+        else y = goalY;
+
+        // don't even bother if its off screen
+        if (y == minY) return;
+
+        String translationKey = block.getTranslationKey();
+        MutableText text = TextUtils.translatable(translationKey);
+        int w = client.textRenderer.getWidth(text);
+
+        // transition the width
+        if (showAnimations) width = MathHelper.lerp(.8f * delta, width, w);
+        else width = w;
+
+        AnchoredBounds bounds = new AnchoredBounds(0, (int)y, (int)width + 20 + padding * 2, height + padding * 2, client.getWindow().getScaledWidth(), client.getWindow().getScaledHeight(), Anchor.TopCenter, Anchor.TopCenter);
+
+        Color backgroundColor = new Color(0, 0, 0, .5f);
+        int backgroundColorInt = backgroundColor.getRGB();
+
+        DrawUtils.drawRect(context, bounds, backgroundColorInt);
+        DrawUtils.drawRect(context, bounds.x - border, bounds.y, border, bounds.height, backgroundColorInt);
+        DrawUtils.drawRect(context, bounds.x + bounds.width, bounds.y, border, bounds.height, backgroundColorInt);
+        DrawUtils.drawRect(context, bounds.x, bounds.y + bounds.height, bounds.width, border, backgroundColorInt);
+
+        context.drawText(client.textRenderer, text, bounds.x + 20 + padding, bounds.y + 5 + padding, 0xFFFFFF, false);
+        itemRender.render(bounds.x + padding,  bounds.y + padding, new ItemStack(block));
+    }
+
+    private Block getBlock() {
+        ClientWorld world = client.world;
+        if (world == null) return null;
+
+        Entity camera = client.getCameraEntity();
+        if (camera == null) return null;
+
+        ClientPlayerInteractionManager interactionManager = client.interactionManager;
+        if (interactionManager == null) return null;
+
+        float maxRange = interactionManager.getReachDistance();
+
+        Vec3d viewVector = camera.getRotationVec(client.getTickDelta());
+        Vec3d startVector = camera.getCameraPosVec(client.getTickDelta());
```

Comment:
I would put the value of `client.getTickDelta()` in a variable to avoid calling it twice